ui/ducks/send/send.js

```diff
@@ -2046,7 +2048,14 @@ export function updateSendAmount(amount) {
     await dispatch(
       addHistoryEntry(`sendFlow - user set amount to ${logAmount}`),
     );
-    await dispatch(actions.updateSendAmount(amount));
+    if (
+      draftTransaction.asset.type === AssetType.NFT &&
+      draftTransaction.asset.details.standard === TokenStandard.ERC1155
+    ) {
+      await dispatch(actions.updateSendAmount(toHex(amount)));
+    } else {
+      await dispatch(actions.updateSendAmount(amount));
+    }
```

Comment:
An alternative idea: The hex conversion could happen by the erc1155 input instead:

https://github.com/MetaMask/metamask-extension/blob/dcf585f0f22a7645103e97d73f66148244322dcd/ui/pages/confirmations/send/send-content/send-amount-row/send-amount-row.component.js#L70

by doing something like:

```ts
onChange={(amount) => this.handleChange(decimalToHex(amount))}
```

So that this send action doesn't need to add erc1155 specific logic.

This would give it the same behavior as [token inputs](https://github.com/MetaMask/metamask-extension/blob/dcf585f0f22a7645103e97d73f66148244322dcd/ui/components/ui/token-input/token-input.component.js#L93) and [native inputs](https://github.com/MetaMask/metamask-extension/blob/dcf585f0f22a7645103e97d73f66148244322dcd/ui/components/app/currency-input/currency-input.js#L108) which also convert to hex before calling back to `updateSendAmount`

Reply:
eemm that's a better idea i agree , implementation just done + test.
thnx @bergeron for you comment